File: bobcat/datetime/datetime10.cc
#include "datetime.ih"

// tm specifies a LOCALTIME time point in the time zone `zone'
// The tm_year field specifies the calendar year, so 2019 or comparable.
//
// tm's wday, yday and isdst are ignored

DateTime::DateTime(TM const &tm, Zone const &zone)
:
    DateTime(zone)                  // initialize a LOCALTIME zone
{
    d_tm = tm;
    d_tm.tm_year -= 1900;           // -1900: required mktime correction 

    d_utcSec = utcFromTM(d_tm);     // update UTC seconds

    assignTM();                     // update d_tm
}

File: bobcat/datetime/datetime10.f
//     struct tm ts = {0, 0, 10, 5, 6, 109, 0, 0, 1};
//      dst and day-of-year fields ignored.
//      ts defines a UTC time point, zoneMinutes is added to obtain local time
inline DateTime::DateTime(TM const &utcTM, int zoneMinutes)
:
    DateTime(utcTM, zoneMinutes, DSTSpec{ false })      // -> 11.cc
{}

